File: Stage21/C/Problem1912.c
#include <stdio.h>

int max(int a, int b) {
    return a > b ? a : b;
}

int main(void) {
    int n;
    scanf("%d", &n);
    
    int arr[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    int cur = arr[0];
    int maxSum = arr[0];

    for (int i = 1; i < n; i++) {
        cur = max(arr[i], cur + arr[i]);
        maxSum = max(maxSum, cur);
    }

    printf("%d\n", maxSum);
    return 0;
}

File: Stage21/Python/Problem1912.py
def main():
    n = int(input())
    arr = list(map(int, input().split()))
    
    cur = arr[0]
    maxSum = arr[0]
    
    for i in range(1, n):
        cur = max(arr[i], cur + arr[i])
        maxSum = max(maxSum, cur)
    
    print(maxSum)
    
if __name__ == "__main__":
    main()